src/pages/index.js

```diff
@@ -70,31 +141,60 @@ function handleImageClick(name, link) {
   newImagePopup.open({ link, name });
 }
 
-const cardSection = new Section(
-  {
-    items: cardData,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.getView();
   cardListEl.prepend(cardElement);
+  console.log(cardData);
+}
+
+function handleDeleteClick(card) {
+  newPopupConfirm.open();
+  newPopupConfirm.setSubmitAction(() => {
+    api.deleteCard(card.id).then(() => {
+      card.deleteCard();
+      newPopupConfirm.close();
+    });
+  });
 }
 
-cardSection.renderItems();
+function handleLikeClick(card) {
+  if (!card.isLiked) {
+    api.likeCard(card.id).then(() => {
+      card.updateLikes(true);
+    });
+  } else {
+    api.dislikeCard(card.id).then(() => {
+      card.updateLikes(false);
+    });
+  }
+}
 
 function handleProfileFormSubmit({ name, description }) {
-  userInfo.setUserInfo({ name, description });
-  newProfilePopup.close();
+  api.updateProfileInfo({ title: name, description }).then(() => {
+    userInfo.setUserInfo({ name, description });
+  });
 }
 
 function handleCardFormSubmit({ title, url }) {
-  renderCard({ name: title, link: url }, cardsWrap);
-  addCardFormValidator.resetValidation();
-  newCardPopup.close();
+  api.addCard({ name: title, link: url }).then((newCardData) => {
+    renderCard(newCardData);
+    addCardFormValidator.resetValidation();
```

Comment:
` addCardFormValidator.resetValidation();` is not needed here because you don't have errors in `then` . The code here is fullfilled only on successful submitting